handlers/producer.go

```diff
@@ -46,7 +84,28 @@ func CreateHandleMessage(conn *memphis.Conn) func(*fiber.Ctx) error {
 				log.Errorf("CreateHandleMessage - handleHeaders: %s", err.Error())
 				return err
 			}
-			err = conn.Produce(stationName, "rest_gateway", message, []memphis.ProducerOpt{memphis.ProducerGenUniqueSuffix()}, []memphis.ProduceOpt{memphis.MsgHeaders(hdrs)})
+			tenantName, username, err := getTenantNameAndUserFromToken(headers["Authorization"])
+			if err != nil {
+				log.Errorf("CreateHandleMessage - getTenantNameAndUserFromToken: %s", err.Error())
+				c.Status(500)
+				return c.JSON(&fiber.Map{
+					"success": false,
+					"error":   err.Error(),
```

Comment:
not relevant to here because this lines should be deleted anyway, but just to make sure that you are remember that in case of 500 status code we don't return the real error but a Server error message